R/MsBackendDataFrame.R

```diff
@@ -264,26 +281,24 @@ setReplaceMethod("spectraData", "MsBackendMemory", function(object, value) {
 })
 
 #' @rdname hidden_aliases
-setMethod("spectraNames", "MsBackendMemory", function(object) {
+setMethod("spectraNames", "MsBackendDataFrame", function(object) {
     rownames(object@spectraData)
 })
 
 #' @rdname hidden_aliases
-setReplaceMethod("spectraNames", "MsBackendMemory", function(object, value) {
+setReplaceMethod("spectraNames", "MsBackendDataFrame", function(object, value) {
     rownames(object@spectraData) <- value
     validObject(object)
     object
 })
 
 #' @rdname hidden_aliases
-setMethod("spectraVariables", "MsBackendMemory", function(object) {
-    ## TODO @jo IMPLEMENT:
-    ## list the colnames of spectraData AND all columns supported by accessor
-    ## methods.
+setMethod("spectraVariables", "MsBackendDataFrame", function(object) {
+    unique(c(names(.SPECTRA_DATA_COLUMNS), colnames(object@spectraData)))
```

Comment:
Why do you take the union of the actual column names and the mandatory ones defined  in `.SPECTRA_DATA_COLUMNS`?

Reply:
Each `MsBackend` **has** to provide the data from the mandatory columns (*via* accessor methods). In addition, the backend has to tell the user what other fields/variables it provides. Imagine you have a `MsBackendMassBank` that uses a SQL backend to retrieve spectra from a database. You want to know what variables/annotations you would get for a spectrum from the database. `spectraVaribles` should then return you the names of the mandatory variables and the ones it provides in addition, hence the union.

Reply:
I think there's some misunderstanding here. You talk about `MsBackend`, I refer to `MsBackendDataFrame`. `.SPECTRA_DATA_COLUMNS` is defined in the later, hence my questions. Are we talking about different things here?

Reply:
We can clarify this on the hangout, if easier.